docs/ruleset_functions/check_purchased_chw_hhw.md

```diff
@@ -0,0 +1,75 @@
+
+        - For each cooling system in HVAC system: `for cooling_system in hvac_sys.cooling_system:`
+
+          - Check if cooling system is connected to any loop in purchased cooling loop array: `if cooling_system.chilled_water_loop in purchased_chw_loop_array`
+
+            - Set purchased cooling flag as True and break logic loop #1: `purchased_chw_hhw_status_dict["PURCHASED_COOLING"] = TRUE, BREAK` (Note XC, under the assumption that there is no orphan hvac system not serving any zones.)
+
+  - Else, external fluid source is heating type: `else:` (Note XC, under the assumption that the type field input is correct, i.e. only three inputs are valid, cooling, hot water or steam)
+
+    - Get fluid loop served by external fluid source: `heating_loop = GET_COMPONENT_BY_ID(external_fluid_source.loop)`
+
+      - Save fluid loop and its child loops in purchased heating loop array: `purchased_hhw_loop_array.append(heating_loop.id), purchased_hhw_loop_array.append(loop.id for loop in heating_loop.child_loops)`
+
+    - For each building segment in RMR (logic loop #2): `for building_segment in RMR...building_segments:`
+
+      - For each HVAC system in building segment: `for hvac_sys in building_segment.heating_ventilation_air_conditioning_systems:`
+
+        - For each heating system in HVAC system: `for heating_system in hvac_sys.heating_system:`
+
+          - Check if heating system is connected to any loop in purchased heating loop array: `if heating_system.hot_water_loop in purchased_hhw_loop_array`
+
+            - Set purchased heating flag as True and break logic loop #2: `purchased_chw_hhw_status_dict["PURCHASED_HEATING"] = TRUE, BREAK`
+
+        - For each preheat system in HVAC system: `for preheat_system in hvac_sys.preheat_system:`
+
+          - Check if heating system is connected to any loop in purchased heating loop array: `if preheat_system.hot_water_loop in purchased_hhw_loop_array`
+
+            - Set purchased heating flag as True and break logic loop #2: `purchased_chw_hhw_status_dict["PURCHASED_HEATING"] = TRUE, BREAK`
+
+      - For each zone in building segment: `for zone in building_segment.zones:`
+
+        - For each terminal in zone: `for terminal in zone.terminals:`
+
+          - Check if zonal reheat is connected to any loop in purchased heating loop array: `if terminal.reheat_from_loop in purchased_hhw_loop_array`
+
+            - Set purchased heating flag as True and break logic loop #2: `purchased_chw_hhw_status_dict["PURCHASED_HEATING"] = TRUE, BREAK`
+
+**Returns** `return purchased_chw_hhw_status_dict`  
+
+**Notes:**
+
+1. For purchased heating, another option is to check if SHW uses purchased heating. If not and RMR has external fluid source, then assume purchased heating is used for space heating.
+
+2. Pending question on baseboard, radiant system, active beams in zone, whether these use reheat_from_loop or is part of an HVAC system, and whether terminal has cooling output.
```

Comment:
Terminal has an input for 'heating_from_loop', since baseboards would be modeled as terminals, my guess is they would use 'heating_from_loop' parameter. I do not see a data element for 'reheat_from_loop'